uaclient/contract.py

```diff
@@ -212,7 +208,7 @@ def update_activity_token(self):
         machine_token = self.cfg.machine_token.get("machineToken")
         machine_id = system.get_machine_id(self.cfg)
 
-        request_data = self._get_activity_info(machine_id)
+        request_data = self._get_activity_info()
```

Comment:
Not necessarily for this PR, but for the overall approach:

I think in this method, lower below, you are saving the response from the UPDATE_ACTIVITY_TOKEN request in a file. As far as I can tell, in the current implementation the endpoint will return the entire payload (with the new activity ID, token and all the new information you send as is). Is this fine? If not, should we change the behavior s.t. we only return the activity identifiers and not the metadata?

Reply:
That's an excellent point! And something nobody else noticed.

I think this behavior is fine since the extra data is minimal and non-sensitive and its presence doesn't hurt the client - we treat that state file as a json blob, so extra fields are ignored.

In the future we hope to improve/organize our on-disk state and at that time we may think about discarding the extra metadata on the client side, but even then I don't think it hurts for the server to send it back.